src/main/java/org/wildfly/security/auth/server/IdentityLocator.java

```diff
@@ -0,0 +1,180 @@
+
+/**
+ * A locator for an identity.
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public final class IdentityLocator {
+    private final String name;
+    private final Principal principal;
+    private final Evidence evidence;
+
+    IdentityLocator(final String name, final Principal principal, final Evidence evidence) {
+        this.name = name;
+        this.principal = principal;
+        this.evidence = evidence;
+    }
+
+    /**
+     * Get the identity name.
+     *
+     * @return the identity name (not {@code null})
+     * @throws IllegalStateException if the name field is not set
+     */
+    public String getName() {
+        final String name = this.name;
+        if (name == null) throw log.locatorFieldNotSet("name");
+        return name;
+    }
+
+    /**
+     * Determine whether a name is set.
+     *
+     * @return {@code true} if a name is set, {@code false} otherwise
+     */
+    public boolean hasName() {
+        return name != null;
+    }
+
+    /**
+     * Get the identity principal.
+     *
+     * @return the identity principal (not {@code null})
+     * @throws IllegalStateException if the principal field is not set
+     */
+    public Principal getPrincipal() {
+        final Principal principal = this.principal;
+        if (principal == null) throw log.locatorFieldNotSet("principal");
+        return principal;
+    }
+
+    /**
+     * Determine whether a principal is set.
+     *
+     * @return {@code true} if a principal is set, {@code false} otherwise
+     */
+    public boolean hasPrincipal() {
+        return principal != null;
+    }
+
+    /**
+     * Get the identity evidence.
+     *
+     * @return the identity evidence (not {@code null})
+     * @throws IllegalStateException if the evidence field is not set
+     */
+    public Evidence getEvidence() {
+        final Evidence evidence = this.evidence;
+        if (evidence == null) throw log.locatorFieldNotSet("evidence");
+        return evidence;
+    }
+
+    /**
+     * Determine whether evidence is set.
+     *
+     * @return {@code true} if evidence is set, {@code false} otherwise
+     */
+    public boolean hasEvidence() {
+        return evidence != null;
+    }
+
+    /**
+     * Shortcut method to construct an identity locator from just a name.
+     *
+     * @param name the name (must not be {@code null})
+     * @return the identity locator (not {@code null})
+     */
+    public static IdentityLocator fromName(String name) {
+        Assert.checkNotNullParam("name", name);
+        return new IdentityLocator(name, null, null);
+    }
+
+    /**
+     * A class for building {@link IdentityLocator} instances.
+     */
+    public static class Builder {
+        private String name;
+        private Principal principal;
+        private Evidence evidence;
+
+        /**
+         * Construct a new, empty instance.
+         */
+        public Builder() {
```

Comment:
I think we are reasonably consistent across the rest of the project and instead of exposing the constructor of the builder the type being built instead has a static builder() method.